allensdk/brain_observatory/behavior/behavior_project_cache/project_apis/data_io/behavior_neuropixels_project_cloud_api.py

```diff
@@ -54,15 +51,21 @@ def get_behavior_session(
                                f"{behavior_session_id} "
                                f" there are {row.shape[0]} entries.")
         row = row.squeeze()
-        ecephys_session_id = int(row.ecephys_session_id)
-
-        row = self._ecephys_session_table.query(f"index=={ecephys_session_id}")
+        if np.isnan(row.ecephys_session_id):
+            ecephys_session_id = None
+        else:
+            ecephys_session_id = int(row.ecephys_session_id)
+            temp_row = self._ecephys_session_table.query(
+                f"index=={ecephys_session_id}")
+            if len(temp_row) != 0:
```

Comment:
this check is not needed since it will be caught below. `temp_row` should be `row` since when there is an ecephys session id but the ecephys session id is not in the ecephys_session table, that would make `row` nonempty but the file id null which would fail on line 70.

Reply:
Recall that there are behavior sessions with no ecephys data. I've added a comment to try to clarify.